WordPress/gutenberg
# Show theme, plugin or author in Added By column with appropriate icon or avatar

## Description
This is the start of improving the 'Added by' column in the Templates / Template Parts listing, as described in this comment - https://github.com/WordPress/gutenberg/issues/36597#issuecomment-973893084

There are associated REST API changes that improve how this works - https://github.com/WordPress/wordpress-develop/pull/1937. This comment explains the differences with and without those changes - https://github.com/WordPress/gutenberg/pull/36763#issuecomment-979706550.

## How has this been tested?
Because template parts and templates currently don't support 'author' the easiest way to test this will be by viewing a list of posts instead of templates:
1. Navigate to http://localhost:8888/wp-admin/themes.php?page=gutenberg-edit-site&postType=post
2. Confirm the 'Added by' column shows avatar and user name like this:
<img width="813" alt="Screenshot 2021-11-23 at 3 22 16 pm" src="https://user-images.githubusercontent.com/677833/142984524-1548c830-36a1-47c9-bbd6-5ab086a48567.png">

### Customization
1. Visit the template or template parts page
2. The added by column should show something like this:
<img width="795" alt="Screenshot 2021-11-23 at 3 24 57 pm" src="https://user-images.githubusercontent.com/677833/142984727-3c805031-f013-48aa-a98c-dc6ce776d5ba.png">
3. Customize the template part and return to the page to view it. It should show a blue dot and a tooltip indicating it's customized.


### Template parts created from scratch
1. Visit the template or template parts page
2. Click Add New
3. Save and return to the template or template part page
4. The site will show that this template was added by the site:
<img width="721" alt="Screenshot 2021-11-25 at 6 43 56 pm" src="https://user-images.githubusercontent.com/677833/143537948-ab1137aa-0230-4147-828c-23d035153de8.png">
5. If you have the core REST API changes checked out and built (https://github.com/WordPress/wordpress-develop/pull/1937), and repeat these steps, the author should be shown like it does for posts:
<img width="813" alt="Screenshot 2021-11-23 at 3 22 16 pm" src="https://user-images.githubusercontent.com/677833/143538067-091effcd-85c9-495e-a44a-0deab45aa239.png">

## Types of changes
Enhancement





## Checklist:
